posts/2022-02-17-beta-release-post.adoc

```diff
@@ -0,0 +1,232 @@
+
+import java.sql.*;
+import java.util.concurrent.CompletionException;
+
+@ContextServiceDefinition(name = "java:app/concurrent/AppContextOnly",
+        propagated = APPLICATION,
+        cleared = { TRANSACTION, SECURITY },
+        unchanged = ALL_REMAINING)
+@ManagedExecutorDefinition(name = "java:module/concurrent/MyExecutor",
+        context = "java:app/concurrent/AppContextOnly",
+        maxAsync = 5)
+public class MyServlet extends HttpServlet {
+    @Resource(name = "java:module/env/concurrent/MyExecutorRef",
+              lookup = "java:module/concurrent/MyExecutor")
+    ManagedExecutorService myExecutor;
+
+    @Inject AccountsBean accounts;
+    ...
+----
+
+With the CDI feature enabled, methods that return a `CompletionStage` or `CompletableFuture` can be designated to run asynchronously by annotating them with `@Asynchronous`.
+
+[source, java]
+----
+import jakarta.enterprise.concurrent.Asynchronous;
+import jakarta.inject.Singleton;
+
+import java.sql.*;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionException;
+
+import javax.naming.InitialContext;
+import javax.naming.NamingException;
+import javax.sql.DataSource;
+
+@Singleton
+public class AccountsBean {
+
+    @Asynchronous(executor = "java:module/concurrent/MyExecutor")
+    CompletableFuture<List<Account>> findOverdue(int minDaysLate) {
+        List<Account> overdueAccounts = new ArrayList<Account>();
+        try {
+            // Resource reference lookup requires context of the application component,
+            DataSource db = InitialContext.doLookup("java:comp/env/jdbc/account-db-ref");
+            ...
+            return Asynchronous.Result.complete(overdueAccounts);
+        } catch (NamingException | SQLException x) {
+            throw new CompletionException(x);
+        }
+    }
+
+    ...
+----
+
+When the application invokes the asynchronous method, the container intercepts the invocation and hands back a completion stage representing its eventual completion, which is arranged by the `ManagedExecutorService`.
+
+[source, java]
+----
+    public void doGet(HttpServletRequest req, HttpServletResponse res) {
+        ...
+        accounts.findOverdue(60).thenAccept(this::sendReminder);
+        ...
+    }
+
+    void sendReminder(List<Account> overdue) {
+        try {
+            // Resource reference lookup requires context of the application component,
+            DataSource db = InitialContext.doLookup("java:comp/env/jdbc/customer-db-ref");
+            ...
+        } catch (NamingException | SQLException x) {
+            throw new CompletionException(x);
+        }
+    }
+----
+
+The Concurrency 3.0 specification hasn't been published yet because Jakarta EE 10 is still under development. However, you can preview a draft of it in asciidoc link:https://github.com/eclipse-ee4j/concurrency-api/blob/2abe74ce30589ef53e248009fcb446491fd42a27/specification/src/main/asciidoc/jakarta-concurrency.adoc[here].
```

Comment:
```suggestion
The Concurrency 3.0 specification hasn't been published yet because Jakarta EE 10 is still under development. However, you can preview a draft of it at the link:https://github.com/eclipse-ee4j/concurrency-api/blob/2abe74ce30589ef53e248009fcb446491fd42a27/specification/src/main/asciidoc/jakarta-concurrency.adoc[concurrency-api GitHub page].
```